### DISCUSSION TITLE
Provider Support Update

### DISCUSSION THREAD

**Discussion by kfcampbell** [integrations/terraform-provider-github#1342] [Announcements]
:wave: Hi! My name is Keegan and I go by @kfcampbell on GitHub. I've worked on the GitHub Terraform provider for around a year and a half now. Since its inception, the provider has seen a lot of use and many contributors: 210 at last count. It's been owned officially by Hashicorp, then transitioned to GitHub unofficially (i.e. maintaining this project has not been a part of anybody's day job).

Throughout its lifespan, the community has been engaged, vocal, and supportive of the project. I'd like to thank everybody who has contributed on issues and pull requests: without you none of this would be viable. 

As of October-ish this year, the provider has transitioned to GitHub's SDK team and we will start to support the provider as we currently do Octokit. That means my excellent colleague @nickfloyd may start to contribute to this repo, and that issue/PR triage should be a bit more timely in the future. 

It also means you may see a little bit of change around the repo: we'll be working on changing things like labels, repo metadata, docs, etc. to be more in line with how things are run over in the [Octokit organization](https://github.com/octokit). We also will probably add a stalebot and pare down some of the currently open issues and PRs. If something you care about gets closed, please reopen it! Hopefully an outcome of this process will be additional dialogue with the community about the needs of the provider. 

We also have big visions for the future of the provider. Stay tuned for more updates in the future, and please don't hesitate to share about your vision for it in our newly-opened discussions.

We are thankful for the amount of support and love the provider has received in the form of community contributions, and we're excited to support the project more in the future!

**Comment by stevehipwell** [comment]
@kfcampbell @nickfloyd please could we have a status update about where this project is? There is a huge backlog of PRs which haven't had any feedback on them and this is causing significant pain for the consumers.

**Comment by ViacheslavKudinov** [comment]
Hi @ashtom
I'm reaching out regarding the **Terraform GitHub provide**r: https://registry.terraform.io/providers/integrations/github/

This project is incredibly valuable for integrating **GitHub** with Terraform, but it appears to have become somewhat orphaned and is in need of more dedicated maintenance and development support.

Given its importance to the community, I was hoping **GitHub** might be able to explore options to help ensure its continued health and vitality. This could perhaps involve allocating some resources, identifying potential internal champions, or assisting in finding new community maintainers.

Any guidance or assistance you could offer would be greatly appreciated.

Thank you for your time and consideration

**Comment by philipbremer** [comment]
Hi @ViacheslavKudinov. Phil here responding on behalf of @ashtom. 

To share some visibility into work within GitHub, we are currently focused on API scaling investments. This has broad benefit across our APIs and SDKs including the Terraform Provider, but the work isn’t tracked in this repo.

Thank you for reaching out and giving us a helpful nudge here. We are exploring our options to ensure project health and we do appreciate your thoughtful suggestions.

**Reply by nikolay** [reply]
Why does a large corporation like GitHub or Microsoft not allocate a small team dedicated to the Terraform or OpenTofu provider? For instance, in our small company, we have hundreds of repositories, and all their configurations are managed using Terraform. No one would prefer to manually assign GitHub organization secrets to repositories, establish consistent labels, enable autolinking, implement branch protection rules, and grant team access. All of these tasks are conveniently and transparently accomplished through Terraform. Essentially, by disregarding this provider, you are adversely affecting your most significant customers.

**Reply by ViacheslavKudinov** [reply]
Hello 
As we all know that Thomas (CEO of GitHub) will leave the company, it could be good to hear from @philipbremer, who will be owning the provider and its feature.
We, as enterprise customer, expect GitHub to provide some IaC solution for their product or let the community do it themselves if the GitHub Terraform provider is not important for GitHub.

**Comment by ViacheslavKudinov** [comment]
Hi @philipbremer 

Thank you so much for this update and for the transparency regarding GitHub's current focus on API improvements. It's really encouraging to hear that this foundational work will have broad benefits, including for the GitHub Terraform provider we all ❤️ love.

While that important internal work progresses, I also want to take a moment to acknowledge and deeply appreciate the ongoing dedication and significant efforts of all the contributors to the GitHub Terraform provider. There's a considerable amount of valuable work being poured into its maintenance, bug fixes, and feature enhancements directly within the repository by passionate community members.

The community truly believes in the continued growth and development of the GitHub provider and is actively working to make it better every day. We would like to see a bright future for it and are hopeful for GitHub's increasing commitment as a company to support its evolution.

We're very grateful that GitHub is exploring options to ensure the project's long-term health and genuinely value your consideration of the suggestions made. We look forward to seeing how these internal improvements and potential future support can further empower the provider's success.

**Comment by diofeher** [comment]
Would GitHub's team be open to voting in some community members to take ownership of the project? Fork is an option, but we would rather concentrate efforts in this repository, as long as GitHub provides some ownership to the community...

I've had an open PR since January, and I've noticed many valuable contributions from community members without any activity from maintainers. I'm confident the community can address these issues; we just need some coordination with your team.

**Reply by thomas-hyland** [reply]
I second this. 

Not sure what's going on behind the scenes over at GitHub but I'd be more than happy to come in as a community developer.

**Reply by diofeher** [reply]
By the way, I had a quick chat with Nick Floyd, and he mentioned they're currently deciding on the process for assigning strong community members to oversee the project. So, I think we just need to stay tuned...

**Reply by ViacheslavKudinov** [reply]
Hi
It could be good to link a new discussion about the provider here as well for traceability https://github.com/integrations/terraform-provider-github/discussions/2792

**Reply by diofeher** [reply]
I was added as a contributor, so I'll be helping to maintain this project, probably starting in the middle of next month.

**Comment by deiga** [comment]
There has been an increase in activity and thus this discussion is no longer relevant :)
